data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -215,28 +275,90 @@ public void testFullScan() {
     Iterable<Record> results = IcebergGenerics.read(sharedTable).build();
 
     Set<Record> expected = Sets.newHashSet();
-    expected.addAll(file1Records);
-    expected.addAll(file2Records);
-    expected.addAll(file3Records);
+    expected.addAll(file1SecondSnapshotRecords);
+    expected.addAll(file2SecondSnapshotRecords);
+    expected.addAll(file3SecondSnapshotRecords);
+
+    Set<Record> records = Sets.newHashSet(results);
+    Assert.assertEquals("Should produce correct number of records",
+        expected.size(), records.size());
+    Assert.assertEquals("Random record set should match",
+        Sets.newHashSet(expected), records);
+    Assert.assertNotNull(records.stream().findFirst().get().getField("id"));
+    Assert.assertNotNull(records.stream().findFirst().get().getField("data"));
+  }
+
+  @Test
+  public void testUnknownSnapshotId() {
+    exceptionRule.expect(IllegalArgumentException.class);
+    exceptionRule.expectMessage(startsWith("Cannot find snapshot with ID "));
```

Comment:
Instead of using `ExpectedException`, we use `AssertHelpers.assertThrows`. That also supports matching the contents of the thrown exception's message.